distribution/oci.go

```diff
@@ -27,19 +26,4 @@ func init() {
 	if err != nil {
 		panic(fmt.Sprintf("Unable to register manifest: %s", err))
 	}
-
-	manifestListFunc := func(b []byte) (distribution.Manifest, distribution.Descriptor, error) {
```

Comment:
@dmcgowan I removed this one (originally added in https://github.com/moby/moby/pull/37359), but couldn't find `ocispec.MediaTypeImageManifest` being registered in the updated github.com/docker/distribution package; was that an omission, or is it being handled in a different way?

Reply:
Looking at this code; does this handle the other (`ocispec.MediaTypeImageManifest`) type? https://github.com/docker/distribution/blob/0d3efadf0154c2b8a4e7b6621fff9809655cc580/manifest/manifestlist/manifestlist.go#L143-L155

Reply:
I see that was added in https://github.com/docker/distribution/commit/c94f28805e1b653204cc4cec2d76a12398e4524a by @mikebrow 

@mikebrow does that mean I can remove https://github.com/moby/moby/pull/38718/files#diff-0c3ca463b55185d48e515450af07fe93R25 now, and it will be handled by that `FromDescriptors` code? (sorry, quite a n00b on that part of the code, so thought I'd ask)

Reply:
OK; getting further; looks like it's registered here; https://github.com/docker/distribution/blob/0d3efadf0154c2b8a4e7b6621fff9809655cc580/manifest/ocischema/manifest.go#L23-L38

But that file doesn't get vendored 🤔

Reply:
https://github.com/docker/distribution/blob/0d3efadf0154c2b8a4e7b6621fff9809655cc580/manifest/manifestlist/manifestlist.go#L73 registers this index. It doesn't look like the ocischema is getting imported anywhere, it it was then it could replace the first one too. Right now the others get pulled in by using those packages but I am not sure this implicit registration approach is great. We are deprecating use of those structures though so it is fine to just leave it be.